File: java/src/main/java/toolgood/flowVision/Flows/AppWork.java
package toolgood.flowVision.Flows;

import com.alibaba.fastjson2.JSONArray;
import com.alibaba.fastjson2.JSONObject;
import toolgood.flowVision.Flows.Enums.CellType;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class AppWork {
    private ProjectWork Project;
    public String Code;
    public String Name;
    public List<AppInitValueWork> InitValueList;
    public List<AppInputWork> InputList;
    public Map<String, NodeWork> AllNodeWork;
    public Map<String, List<NodeWork>> ParentNodeWorks;
    public StartFlowWork Start;

    public void Init(ProjectWork work) {
        Project = work;
        for (int i = 0; i < InitValueList.size(); i++) {
            AppInitValueWork item = InitValueList.get(i);
            item.Init(work);
        }
        for (int i = 0; i < InputList.size(); i++) {
            AppInputWork item = InputList.get(i);
            item.Init(work);
        }
        for (NodeWork item : AllNodeWork.values()) {
            item.Init(work, this);
            if (item.NodeType == CellType.Start) {
                Start = (StartFlowWork) item;
            }
        }

        ParentNodeWorks = new HashMap<>();
        for (NodeWork item : AllNodeWork.values()) {
            for (List<NodeWork> nextNode : item.NextNodes.values()) {
                for (int i = 0; i < nextNode.size(); i++) {
                    NodeWork node = nextNode.get(i);
                    if (node.NodeType == CellType.Procedure) {
                        List<NodeWork> list;
                        if (ParentNodeWorks.containsKey(node.Id)) {
                            list = ParentNodeWorks.get(node.Id);
                            list.add(item);
                        } else {
                            list = new ArrayList<>();
                            ParentNodeWorks.put(node.Id, list);
                        }
                    }
                }
            }
        }


    }

    final static AppWork parse(JSONObject jsonObject) throws Exception {
        AppWork result = new AppWork();
        result.Code = jsonObject.getString("code");
        result.Name = jsonObject.getString("name");

        result.InitValueList = new ArrayList<>();
        JSONArray initValueList = jsonObject.getJSONArray("initValueList");
        for (Object item : initValueList) {
            if (item instanceof JSONObject jsonObject1) {
                AppInitValueWork work = AppInitValueWork.parse(jsonObject1);
                result.InitValueList.add(work);
            }
        }

        result.InputList = new ArrayList<>();
        JSONArray inputList = jsonObject.getJSONArray("inputList");
        for (Object item : inputList) {
            if (item instanceof JSONObject jsonObject1) {
                AppInputWork work = AppInputWork.parse(jsonObject1);
                result.InputList.add(work);
            }
        }

        result.AllNodeWork = new HashMap<>();
        JSONObject allNodeWork = jsonObject.getJSONObject("allNodeWork");
        for (Map.Entry<String, Object> item : allNodeWork.entrySet()) {
            if (item.getValue() instanceof JSONObject jsonObject1) {
                NodeWork work = NodeWork.parse(jsonObject1);
                result.AllNodeWork.put(item.getKey(), work);
            }
        }

        return result;
    }
}

File: editor/ToolGood.FlowWork.Applications/Flows/AppWork.cs
using System.Text.Json.Serialization;

namespace ToolGood.FlowWork.Flows
{
	public sealed class AppWork
	{
		/// <summary>
		/// 编码
		/// </summary>
		[System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
		public string Code { get; set; }

		/// <summary>
		/// 名称
		/// </summary>
		[System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
		public string Name { get; set; }

		public List<AppInitValueWork> InitValueList { get; set; }
		public List<AppInputWork> InputList { get; set; }
		public Dictionary<string, NodeWork> AllNodeWork { get; set; }

		[System.Text.Json.Serialization.JsonIgnore]
		public Dictionary<string, List<NodeWork>> ParentNodeWorks { get; set; }

		[System.Text.Json.Serialization.JsonIgnore]
		public StartFlowWork Start { get; set; }

		internal void Init(ProjectWork work)
		{
			for (int i = 0; i < InitValueList.Count; i++) {
				var item = InitValueList[i];
				item.Init(work);
			}
			for (int i = 0; i < InputList.Count; i++) {
				var item = InputList[i];
				item.Init(work);
			}
			foreach (var item in AllNodeWork) {
				item.Value.Init(work, this);
			}
			Start = AllNodeWork.FirstOrDefault(q => q.Value.NodeType == CellType.Start).Value as StartFlowWork;

			ParentNodeWorks = new Dictionary<string, List<NodeWork>>();
			foreach (var item in AllNodeWork) {
				foreach (var nextNode in item.Value.NextNodes) {
					for (int i = 0; i < nextNode.Value.Count; i++) {
						var node = nextNode.Value[i];
						if (node is ProcedureFlowWork) {
							List<NodeWork> list;
							if (ParentNodeWorks.TryGetValue(node.Id, out list) == false) {
								list = new List<NodeWork>();
								ParentNodeWorks[node.Id] = list;
							}
							list.Add(item.Value);
						}
					}
				}
			}
		}
	}
}